core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -148,6 +149,7 @@
 
     // must initialized in doStart(), when we have the allocationService set
     private volatile NodeJoinController nodeJoinController;
+    private volatile NodeRemovalClusterStateTaskExecutor nodeRemovalExecutor;
```

Comment:
but `electMasterService` is passed to the ZenDiscovery constructor? I don't follow, sorry.